source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -212,51 +211,27 @@ CIrrDeviceSDL::CIrrDeviceSDL(const SIrrlichtCreationParameters& param)
 		}
 
 #if defined(_IRR_WINDOWS_)
-		SDL_putenv("SDL_VIDEODRIVER=directx");
+		SDL_setenv("SDL_VIDEODRIVER","directx", 1);
 #elif defined(_IRR_OSX_PLATFORM_)
-		SDL_putenv("SDL_VIDEODRIVER=Quartz");
+		SDL_setenv("SDL_VIDEODRIVER", "Quartz", 1);
 #elif !defined(_IRR_EMSCRIPTEN_PLATFORM_)
-		SDL_putenv("SDL_VIDEODRIVER=x11");
+		SDL_setenv("SDL_VIDEODRIVER", "x11", 1);
 #endif
 	}
 
 //	SDL_putenv("SDL_WINDOWID=");
 
-	SDL_VERSION(&Info.version);
-
-#ifndef _IRR_EMSCRIPTEN_PLATFORM_
-	SDL_GetWMInfo(&Info);
-#endif //_IRR_EMSCRIPTEN_PLATFORM_
-	core::stringc sdlversion = "SDL Version ";
-	sdlversion += Info.version.major;
-	sdlversion += ".";
-	sdlversion += Info.version.minor;
-	sdlversion += ".";
-	sdlversion += Info.version.patch;
-
-	Operator = new COSOperator(sdlversion);
-	if ( SDLDeviceInstances == 1 )
-	{
-		os::Printer::log(sdlversion.c_str(), ELL_INFORMATION);
-	}
-
 	// create keymap
 	createKeyMap();
-	// enable key to character translation
-	SDL_EnableUNICODE(1);
-
-	(void)SDL_EnableKeyRepeat(SDL_DEFAULT_REPEAT_DELAY, SDL_DEFAULT_REPEAT_INTERVAL);
 
 	if ( CreationParams.Fullscreen )
-		SDL_Flags |= SDL_FULLSCREEN;
+		SDL_Flags |= SDL_WINDOW_FULLSCREEN;
 	else if ( Resizable )
-		SDL_Flags |= SDL_RESIZABLE;
+		SDL_Flags |= SDL_WINDOW_RESIZABLE;
 	if (CreationParams.DriverType == video::EDT_OPENGL)
-		SDL_Flags |= SDL_OPENGL;
-	else if (CreationParams.Doublebuffer)

```

Comment:
should the doublebuffer option result in a call `SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 0)` (or similar) or is that already the default?

Reply:
SDL2 by default is double buffered. SDL_GL_DOUBLEBUFFER is boolean, so I guess I should set the attribute to false if single buffering is required